src/lib/generator/writeJson.js

```diff
@@ -1,38 +1,66 @@
 import fs from 'fs';
 import _ from 'lodash';
-import { join, dirname, resolve } from 'path';
-import { parseRow, formatIcdWithPeriod, isValidIcdLength } from '$lib/utils';
 
-const __dirname = resolve(dirname(''));
+import { parseRow, formatIcdWithPeriod } from '$lib/utils';
+
+import { icdCodesJsonFilePath, hccMappingsFilePath, icdDescriptionsFilePath } from '.';
 
 /**
  * Parses a text file content into an array of objects containing code and description.
  * @param {string} textFile - The content of the text file as a string.
- * @returns {Array<App.CodeEntry>} An array of objects containing code and description extracted from each line of the text file.
+ * @returns {Record<string, App.CodeEntry>} An object keyed by code containing code and description extracted from each line of the text file.
  */
-function parseTextFile(textFile) {
-    return _.map(textFile.split('\r\n'), (string) => {
-        const spaceIndex = string.indexOf(' ');
-        const code = string.slice(0, spaceIndex).trim();
-        return {
-            code,
-            description: string.slice(spaceIndex + 1).trim(),
-        };
-    });
+function parseDescriptionFile(textFile) {
+    return _.reduce(textFile.split(/\r\n|\n|\r/), (memo, string) => {
+        const code = formatIcdWithPeriod(string.slice(0, 8));
+        const description = string.slice(8);
+
+        if (!code) return memo;
+
+        // @ts-ignore
+        memo[code] = { code, description, is_specific: true };
+
+        return memo;
+    }, {});
 }
 
 /**
  * Parses a CSV file content into an array of objects containing code, description, and additional details.
  * @param {string} csvFile - The content of the CSV file as a string.
- * @returns {Array<App.CodeEntry>} An array of objects containing code, description, and additional details extracted from each line of the CSV file.
+ * @param {Record<string, App.CodeEntry>} icdList - The list of ICD codes from the descriptions file
+ * @returns {Record<string, App.CodeEntry>} An object keyed by code containing code, description, and additional details extracted from each line of the CSV file.
  */
-function parseCsvFile(csvFile) {
+function parseHccMapping(csvFile, icdList) {
+    const DIVIDER = ',,,,,,,,,,,,,,,';
+    let parseHeader = false;
+    let parseHeaderRow = false;
+    let parseFooter = false;
+
     return _.reduce(
-        csvFile.split('\r\n'),
+        csvFile.split(/\r\n|\n|\r/),
         (memo, string) => {
+            if (!parseFooter && string === DIVIDER) {
```

Comment:
This bit is the craziest and most likely to break for future files.. I'm really not sure what to do about it..  We might have to update this to support new formats if new columns are added, although I think all it will require is updating the `DIVIDER`. I _thought_ about saving the divider when I got to the first line that starts with `,,,,` or something like that.. might be better future proof, but really.. who knows..

Reply:
I ended up modifying this to store the column count of the divider for the check in the `!parseHeaderRow`